modules/apps/portal-search/portal-search-rest-test/src/testIntegration/java/com/liferay/portal/search/rest/resource/v1_0/test/SearchResultResourceTest.java

```diff
@@ -1131,6 +1145,91 @@ private void _testPostSearchPageWithKeywords() throws Exception {
 		Assert.assertEquals(1L, searchPage.getTotalCount());
 	}
 
+	private void _testPostSearchPageWithLocalizedTextFields() throws Exception {
+		PrincipalThreadLocal.setName(_user.getUserId());
+		PermissionThreadLocal.setPermissionChecker(
+			PermissionCheckerFactoryUtil.create(_user));
+
+		String objectFieldName = "localized" + RandomTestUtil.randomString();
+
+		_objectDefinition =
+			_objectDefinitionLocalService.addCustomObjectDefinition(
+				_user.getUserId(), 0, null, false, true, true, false,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				ObjectDefinitionTestUtil.getRandomName(), null, null,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				true, ObjectDefinitionConstants.SCOPE_COMPANY,
+				ObjectDefinitionConstants.STORAGE_TYPE_DEFAULT, null);
+
+		ObjectFieldUtil.addCustomObjectField(
+			new TextObjectFieldBuilder(
+			).userId(
+				_user.getUserId()
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).indexed(
+				true
+			).name(
+				objectFieldName
+			).objectDefinitionId(
+				_objectDefinition.getObjectDefinitionId()
+			).localized(
+				true
+			).build());
+
+		ObjectFieldUtil.addCustomObjectField(
+			new TextObjectFieldBuilder(
+			).userId(
+				_user.getUserId()
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).indexed(
+				true
+			).name(
+				"foo"
+			).objectDefinitionId(
+				_objectDefinition.getObjectDefinitionId()
+			).build());
+
+		_objectDefinition =
+			_objectDefinitionLocalService.publishCustomObjectDefinition(
+				-_user.getUserId(), _objectDefinition.getObjectDefinitionId());
+
+		_defaultObjectEntryManager =
+			(DefaultObjectEntryManager)_objectEntryManager;
+
+		_dtoConverterContext = new DefaultDTOConverterContext(
+			false, Collections.emptyMap(), _dtoConverterRegistry, null,
+			LocaleUtil.getDefault(), null, TestPropsValues.getUser());
+
+		_defaultObjectEntryManager.addObjectEntry(
+			_dtoConverterContext, _objectDefinition,
+			new ObjectEntry() {
+				{
+					properties = HashMapBuilder.<String, Object>put(
+						objectFieldName + "_i18n",
+						HashMapBuilder.put(
+							"en_US", "localizedEntry1"
+						).put(
+							"pt_BR", "localizedEntry2"
+						).build()
+					).put(
+						"foo", "fooValue"
+					).build();
+				}
+			},
+			ObjectDefinitionConstants.SCOPE_COMPANY);
+
+		SearchPage<SearchResult> searchPage = _postSearchPage(
+			_objectDefinition.getClassName(), null, "localizedEntry1",
```

Comment:
@JhossephG I believe we should search by `pt_BR` because this way we are making sure that the behavior is the expected, otherwise this test will not cover the use case.